lib/InfluxRequest.js

```diff
@@ -106,20 +107,23 @@ InfluxRequest.prototype._request = function (options, callback) {
   // need to store the original path, in case we need to re-submit the request onError and change the host
   if (!requestOptions.originalUrl) requestOptions.originalUrl = requestOptions.url
   requestOptions.url = this.url(host, requestOptions.originalUrl)
-  requestOptions.retries++
-  request(requestOptions, function (err, response, body) {
-    self._parseCallback(err, response, body, requestOptions, callback)
-  })
-}
-
-InfluxRequest.prototype._parseCallback = function (err, response, body, requestOptions, callback) {
-  if (err && resubmitErrorCodes.indexOf(err.code) !== -1) {
-    this.disableHost(requestOptions.host)
-    if (this.options.maxRetries >= requestOptions.retries && this.hostIsAvailable()) {
-      return this._request(requestOptions, callback)
+  requestOptions.retries++  // fetch() ignores the `.retries` option
+  var theResponse
+  fetch(requestOptions.url, requestOptions).then(function (response) {
+    theResponse = response
+    return response.text()
+  }).then(function (body) {
+    body = body === '' ? {} : JSON.parse(body)  // workaround for https://github.com/bitinn/node-fetch/issues/165
```

Comment:
Perhaps checking the statusCode for 204s would be cleaner here? I believe the issue on `node-fetch` would break away from the [whatwg specification](https://fetch.spec.whatwg.org/), which simply states that `response.json()` should

> [r]eturn the result of invoking the initial value of the parse property of the JSON object with the result of running utf-8 decode on bytes as argument. Rethrow any exceptions.